src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -39,9 +38,9 @@ public PoPAuthenticationScheme(PoPAuthenticationConfiguration popAuthenticationC
 
             _popAuthenticationConfiguration = popAuthenticationConfiguration ?? throw new ArgumentNullException(nameof(popAuthenticationConfiguration));
 
-            _popAuthenticationConfiguration.PopCryptoProvider = _popAuthenticationConfiguration.PopCryptoProvider ?? serviceBundle.PlatformProxy.GetDefaultPoPCryptoProvider();
+            _popCryptoProvider = _popAuthenticationConfiguration.PopCryptoProvider ?? serviceBundle.PlatformProxy.GetDefaultPoPCryptoProvider();
```

Comment:
Turns out there was actually a bug here for the default provider. Once the '_popAuthenticationConfiguration.PopCryptoProvider' was set with the default provider, the 'GetDefaultPoPCryptoProvider()' would never be called again for the application instance and the `keyID` would not update.

@bgavrilMS

Reply:
Yeah, good unit testing has the tendency to spot issues.

Reply:
@trwalke @bgavrilMS This is a bug in non-test related code right? Should we create a separate issue for this to reference in release notes?

Reply:
@pmaytak makes sense